src/Common/src/CoreLib/System/ReadOnlySpan.cs

```diff
@@ -24,227 +19,38 @@ namespace System
-        /// </exception>
-        /// <exception cref="System.ArgumentOutOfRangeException">
-        /// Thrown when the specified <paramref name="length"/> is negative.
-        /// </exception>
-        [CLSCompliant(false)]
-        [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        public unsafe ReadOnlySpan(void* pointer, int length)
-        {
-            if (RuntimeHelpers.IsReferenceOrContainsReferences<T>())
-                ThrowHelper.ThrowInvalidTypeWithPointersNotSupported(typeof(T));
-            if (length < 0)
-                ThrowHelper.ThrowArgumentOutOfRangeException();
-
-            _pointer = new ByReference<T>(ref Unsafe.As<byte, T>(ref *(byte*)pointer));
-            _length = length;
-        }
-
-        /// <summary>
-        /// Create a new read-only span over a portion of a regular managed object. This can be useful
-        /// if part of a managed object represents a "fixed array." This is dangerous because neither the
-        /// <paramref name="length"/> is checked, nor <paramref name="obj"/> being null, nor the fact that
-        /// "rawPointer" actually lies within <paramref name="obj"/>.
-        /// </summary>
-        /// <param name="obj">The managed object that contains the data to span over.</param>
-        /// <param name="objectData">A reference to data within that object.</param>
-        /// <param name="length">The number of <typeparamref name="T"/> elements the memory contains.</param>
-        [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        [EditorBrowsable(EditorBrowsableState.Never)]
-        public static ReadOnlySpan<T> DangerousCreate(object obj, ref T objectData, int length) => new ReadOnlySpan<T>(ref objectData, length);
-
-        // Constructor for internal use only.
-        [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        internal ReadOnlySpan(ref T ptr, int length)
-        {
-            Debug.Assert(length >= 0);
-
-            _pointer = new ByReference<T>(ref ptr);
-            _length = length;
-        }
-
         //Debugger Display = {T[length]}
         private string DebuggerDisplay => string.Format("{{{0}[{1}]}}", typeof(T).Name, _length);
 
-        /// <summary>
-        /// Returns a reference to the 0th element of the Span. If the Span is empty, returns a reference to the location where the 0th element
-        /// would have been stored. Such a reference can be used for pinning but must never be dereferenced.
-        /// </summary>
-        [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        [EditorBrowsable(EditorBrowsableState.Never)]
-        internal ref readonly T DangerousGetPinnableReference()
-        {
-            return ref _pointer.Value;
-        }
-
         /// <summary>
         /// The number of items in the read-only span.
         /// </summary>
         public int Length
         {
+#if !FEATURE_PORTABLE_SPAN
```

Comment:
3 ifdefs to shared 150 files. Is it worth it?

Reply:
Any future shared code will go to this shared files and hopefully, down the road, we may be able to get rid of the #if's.

Reply:
Down the road, we will likely stop building the slow Span. System.Memory is partial OOB. Once it ships, the surface of the portable version will be locked down. We won't be able to add to it.

Reply:
I am fine if you think it is not worth it to split the files and I can close this PR. I don't have a strong feeling about it either. @ahsonkhan asked if we can refactor the common code to the shared place. so, I am fine either way :-)

Reply:
> I don't have a strong feeling about it either.

Same here.

Reply:
ok, as I have the changes ready and passing the CI, I'll continue with this PR. let me know if you have any more comments before I merge it. thanks for your comments.

Reply:
> @ahsonkhan asked if we can refactor the common code to the shared place. so, I am fine either way :-)

I think the enumerator, at the very least, should be shared, since it is already contained enough that it can be extracted out easily.

> Down the road, we will likely stop building the slow Span. System.Memory is partial OOB. 

Given this, extracting the other common code is less important, but I think it makes maintainability a bit easier.